### DISCUSSION TITLE
python: Is there a way to append to the mcap file?

### DISCUSSION THREAD

**Discussion by asd12312394** [foxglove/mcap#1154] [General]
As title
 Is there a way to append to the mcap file? 
in Python mcap packages

**Comment by jtbandes** [comment]
Append functionality is not provided in the python library currently. (We have two implementations of this now, for [Go (part of the CLI only)](https://github.com/foxglove/mcap/blob/40cccdfeec54c888397a9941943ce79a1d9d5267/go/cli/mcap/utils/mcap_amendment.go#L26C6-L26C15) and [TypeScript](https://github.com/foxglove/mcap/blob/40cccdfeec54c888397a9941943ce79a1d9d5267/typescript/core/src/McapWriter.ts#L132-L142).) Instead you can try:
- write new data to a separate file and use `mcap merge` from the [CLI tool](https://mcap.dev/guides/cli) to merge the files
- create a reader to read the old file, write the data into a new file, and then continue writing additional data with the same writer.

**Reply by OpenGHz** [reply]
How is it now?

**Reply by jtbandes** [reply]
No change as far as I know.
